### PR TITLE
chore: add pluming necessary for testing CreateChannel

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Adds an event for channel creation and the necessary functions/functionality in `testing/`. Note to reviewers: this will move to `channel/v2` after tests have run successfully using packetserver.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/packet-server/types/events.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	ibcexported "github.com/cosmos/ibc-go/v9/modules/core/exported"
+)
+
+// IBC channel events
+const (
+	AttributeKeyChannelID = "channel_id"
+)
+// IBC channel events vars
+var (
+	EventTypeCreateChannel = "create_channel"
```
Comment (bznein) [inline modules/core/packet-server/types/events.go:15]: any reason this is not a constant?

File: testing/path.go [hunk 2]
```diff
@@ -152,6 +152,9 @@ func (path *Path) Setup() {
 // This is all that is necessary for path setup with the Eureka (V2) protocol
 func (path *Path) SetupV2() {
 	path.SetupClients()
+
+	// path.CreateChannelsV2()
```
Comment (bznein) [inline testing/path.go:156]: leftover comment?

**Review by bznein — APPROVED** [review]

File: testing/endpoint.go [hunk 3]
```diff
@@ -188,6 +188,24 @@ func (endpoint *Endpoint) ProvideCounterparty() (err error) {
 	return err
 }
 
+// CreateChannel will construct and execute a new MsgCreateChannel on the associated endpoint.
+func (endpoint *Endpoint) CreateChannel() (err error) {
+	merklePath := commitmenttypes.NewMerklePath([]byte("ibc"), []byte(""))
```
Comment (damiannolan) [inline testing/endpoint.go:193]: We could probably make `[]byte("")` a var in testing values

File: testing/events.go [hunk 4]
```diff
@@ -56,6 +57,19 @@ func ParseChannelIDFromEvents(events []abci.Event) (string, error) {
 	return "", errors.New("channel identifier event attribute not found")
 }
 
+// ParseChannelIDV2FromEvents parses events emitted from a CreateChannel
+// and returns the channel identifier.
+func ParseChannelIDV2FromEvents(events []abci.Event) (string, error) {
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply testing/events.go:62]: Yeah I'd prefer to not have the V2 suffix in func names and stuff, personally

File: testing/path.go [hunk 5]
```diff
@@ -252,6 +255,19 @@ func (path *Path) CreateChannels() {
 	}
 }
 
+// CreateChannelsV2 initializes two channel endpoints by executing CreateChannel on both chainA and chainB.
+func (path *Path) CreateChannelsV2() {
```
Comment (damiannolan) [inline testing/path.go:259]: I guess there's no room to get rid of V2 here?

**Review by damiannolan — APPROVED** [review]
Nice one @DimitrisJim

**Review by AdityaSripal — APPROVED** [review]

File: modules/core/packet-server/types/events.go (hunk 1 above)
Comment (chatton) [reply modules/core/packet-server/types/events.go:16]: feel like a const makes more sense here, we can worry about it when we move stuff over to msg_server_v2_land anyway

**Review by chatton — APPROVED** [review]
LGTM! Nice work,

**Pull request merged**